src/serial/serialcommands.cpp

```diff
@@ -37,19 +38,37 @@ namespace SerialCommands {
 
     CmdCallback<6> cmdCallbacks;
     CmdParser cmdParser;
-    CmdBuffer<64> cmdBuffer;
+    CmdBuffer<256> cmdBuffer;
 
     void cmdSet(CmdParser * parser) {
-        if(parser->getParamCount() != 1 && parser->equalCmdParam(1, "WIFI")  ) {
-            if(parser->getParamCount() < 3) {
-                logger.error("CMD SET WIFI ERROR: Too few arguments");
-                logger.info("Syntax: SET WIFI \"<SSID>\" \"<PASSWORD>\"");
-            } else {
-                WiFiNetwork::setWiFiCredentials(parser->getCmdParam(2), parser->getCmdParam(3));
-                logger.info("CMD SET WIFI OK: New wifi credentials set, reconnecting");
-            }
+        if(parser->getParamCount() != 1) {
+			if (parser->equalCmdParam(1, "WIFI")) {
+				if(parser->getParamCount() < 3) {
+					logger.error("CMD SET WIFI ERROR: Too few arguments");
+					logger.info("Syntax: SET WIFI \"<SSID>\" \"<PASSWORD>\"");
+				} else {
+					WiFiNetwork::setWiFiCredentials(parser->getCmdParam(2), parser->getCmdParam(3));
+					logger.info("CMD SET WIFI OK: New wifi credentials set, reconnecting");
+				}
+			} else if (parser->equalCmdParam(1, "BWIFI")) {
+				if(parser->getParamCount() < 3) {
+					logger.error("CMD SET BWIFI ERROR: Too few arguments");
+					logger.info("Syntax: SET BWIFI <B64SSID> <B64PASSWORD>");
+				} else {
+					char ssid[33];
+					char pass[65];
+					const char * b64ssid = parser->getCmdParam(2);
+					const char * b64pass = parser->getCmdParam(3);
+					base64_decode_chars(b64ssid, strlen_P(b64ssid), ssid);
```

Comment:
Is this conversation length save?  so ssid will not be writen longer that it is?